graylog2-server/src/main/java/org/graylog2/rest/resources/system/ldap/LdapResource.java

```diff
@@ -298,15 +298,13 @@ public Response updateGroupMappingSettings(@ApiParam(name = "JSON body", require
             config.setCredentials(ldapSettings.getSystemPassword());
         }
 
-        try {
-            LdapNetworkConnection connection = ldapConnector.connect(config);
-            final Set<String> groups = ldapConnector.listGroups(connection,
-                                                                ldapSettings.getGroupSearchBase(),
-                                                                ldapSettings.getGroupSearchPattern(),
-                                                                ldapSettings.getGroupIdAttribute()
+        try (LdapNetworkConnection connection = ldapConnector.connect(config)) {
+            return ldapConnector.listGroups(connection,
+                                            ldapSettings.getGroupSearchBase(),
+                                            ldapSettings.getGroupSearchPattern(),
+                                            ldapSettings.getGroupIdAttribute()
             );
-            return groups;
-        } catch (LdapException e) {
+        } catch (IOException | LdapException e) {
             LOG.error("Unable to retrieve available LDAP groups", e);
             throw new InternalServerErrorException(e);
```

Comment:
If the exception contained a message, the user would see it on the REST API and could act accordingly. This way only a generic HTTP Status 500 response is being generated (also see our `WebApplicationExceptionMapper`).